### PR TITLE
[Statsd receiver]Add metric type as a dimension

### PR DESCRIPTION
**Pull request by JohnWu20**
**Description:** 

Add a new dimension for the metric_type for statsd receiver.

Add a new flag label for the config file of the statsd.  `metric_type_enable`

**Testing:** 
Unit test and local test.

### CODE REVIEW

File: exporter/lokiexporter/logproto/logproto.pb.go [hunk 1]
```diff
@@ -2492,7 +2492,7 @@ func (this *SeriesIdentifier) GoString() string {
 	s := make([]string, 0, 5)
 	s = append(s, "&logproto.SeriesIdentifier{")
 	keysForLabels := make([]string, 0, len(this.Labels))
-	for k, _ := range this.Labels {
```
Comment (gavindoudou) [inline exporter/lokiexporter/logproto/logproto.pb.go:2495]: Why do we need to change the exporter?

Comment (JohnWu20) [reply exporter/lokiexporter/logproto/logproto.pb.go:2495]: There is something wrong with the code format of this exporter, I just ran `make fmt`, this one changed.

Comment (gavindoudou) [reply exporter/lokiexporter/logproto/logproto.pb.go:2495]: I don't think we can revise this part

File: receiver/statsdreceiver/protocol/statsd_parser_test.go [hunk 2]
```diff
@@ -102,16 +112,23 @@ func Test_ParseMessageToMetric(t *testing.T) {
...
 					},
+					{
+						Value:    "counter",
+						HasValue: true,
+					},
 				}),
 		},
 		{
-			name:  "counter metric with sample rate(not divisible) and tag",
+			name:  "counter metric with sample rate(not divisible) and tag", ///
```
Comment (gavindoudou) [inline receiver/statsdreceiver/protocol/statsd_parser_test.go:131]: Need to delete `///`

File: receiver/statsdreceiver/protocol/statsd_parser_test.go [hunk 3]
```diff
@@ -88,7 +98,7 @@ func Test_ParseMessageToMetric(t *testing.T) {
 			err:   errors.New("unsupported metric type: unhandled_type"),
 		},
 		{
-			name:  "counter metric with sample rate and tag",
+			name:  "counter metric with sample rate and tag", ///
```
Comment (gavindoudou) [inline receiver/statsdreceiver/protocol/statsd_parser_test.go:101]: Need to delete `///`

**Comment by gavindoudou** [comment]
Do we have any tests with `metric_type_enable=false`?

File: receiver/statsdreceiver/receiver.go [hunk 4]
```diff
@@ -117,7 +117,7 @@ func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
 						r.Flush(ctx, metrics, r.nextConsumer)
 					}
 				case rawMetric := <-transferChan:
-					r.parser.Aggregate(rawMetric)
+					r.parser.Aggregate(rawMetric, r.config.EnableMetricType)
```
Comment (gavindoudou) [inline receiver/statsdreceiver/receiver.go:120]: For this part, it looks wired to parse the config flag to aggregate each time we receive the metric. The config setup needs to be done before each metric comes. For example, we could have a `getEnableMetricTypeFlag` function to do the setup.

Comment (JohnWu20) [reply receiver/statsdreceiver/receiver.go:120]: Good point, it makes sense to make this value initialized in the config setup part, I will change the code.

File: exporter/lokiexporter/logproto/logproto.pb.go (hunk 1 above)
Comment (JohnWu20) [reply exporter/lokiexporter/logproto/logproto.pb.go:2495]: Will delete this part, it seems that someone else changed it, makes it a conflicting file.

File: receiver/statsdreceiver/protocol/statsd_parser_test.go (hunk 2 above)
Comment (JohnWu20) [reply receiver/statsdreceiver/protocol/statsd_parser_test.go:131]: Sure

File: receiver/statsdreceiver/protocol/statsd_parser_test.go (hunk 3 above)
Comment (JohnWu20) [reply receiver/statsdreceiver/protocol/statsd_parser_test.go:101]: Sure, have no idea how it added.

**Pull request closed without merging**